os/board/rtl8721csm/src/component/common/api/wifi_interactive_mode.c

```diff
@@ -803,72 +803,76 @@ extern int8_t wifi_scan_result_callback(wifi_utils_scan_list_s *scan_list, int s
 
 
 #if SCAN_WITH_SSID
-static void cmd_wifi_scan_with_ssid(int argc, char **argv)
+#define SCAN_BUF_LEN 1000
+int8_t cmd_wifi_scan_with_ssid(trwifi_scan_config_s *scan_config)
 {
 
 	u8 *channel_list = NULL;
 	u8 *pscan_config = NULL;
 
 	char *ssid = NULL;
 	int ssid_len = 0;
-	//Fully scan
-	int scan_buf_len = 500;
-	if (argc == 3 && argv[1] && argv[2]) {
-		ssid = argv[1];
-		ssid_len = strlen((const char *)argv[1]);
-		if ((ssid_len <= 0) || (ssid_len > 32)) {
-			ndbg("\n\rWrong ssid. Length must be less than 32.");
-			goto exit;
+	unsigned char mode = 0;
+	int num_channel = 0;
+		
+	int scan_buf_len = SCAN_BUF_LEN;
+	mode = scan_config->scan_mode;
+
+	if (mode == 0) {	//full scan
+		if (scan_config->ssid != NULL) {
+			ssid = scan_config->ssid;
+			ssid_len = strlen((const char *)ssid);
+			if ((ssid_len <= 0) || (ssid_len > 32)) {
+				RTW_API_INFO("\n\rWrong ssid. Length must be less than 32.");
+				goto exit;
+			}
 		}
-		scan_buf_len = atoi(argv[2]);
-		if (scan_buf_len < 36) {
-			ndbg("\n\rBUFFER_LENGTH too short\n\r");
+		if(scan_buf_len < 36){

```

Comment:
Could you add whitespaces after "if" and ")" please like all other if constructs

Reply:
Done